src/tools/rbd_mirror/PoolWatcher.cc

```diff
@@ -495,12 +563,37 @@ template <typename I>
 void PoolWatcher<I>::notify_listener() {
   dout(10) << dendl;
 
+  std::string mirror_uuid;
   ImageIds added_image_ids;
   ImageIds removed_image_ids;
   {
     Mutex::Locker locker(m_lock);
     assert(m_notify_listener_in_progress);
 
+    // if the mirror uuid is updated, treat it as the removal of all
+    // images in the pool
+    if (m_mirror_uuid != m_pending_mirror_uuid) {
+      if (!m_mirror_uuid.empty()) {
+        dout(0) << "mirror uuid updated:"
+                << "old=" << m_mirror_uuid << ", "
+                << "new=" << m_pending_mirror_uuid << dendl;
+      }
+
+      mirror_uuid = m_mirror_uuid;
+      removed_image_ids = std::move(m_image_ids);
+      m_image_ids.clear();
```

Comment:
@trociny Per the standard, the object is left in an "undefined" state after a move and you are required to put it back into a well-defined state. For STL collections, the "clear" method is used to accomplish that.